### DISCUSSION TITLE
New method "resetExcept"

### DISCUSSION THREAD

**Discussion by SebastianSchoeps** [livewire/livewire#4097] [3. Feature ideas]
If you have a component with many public properties and want to reset all except some, you would have to write all properties into the reset method - and if you add more public properties, you would always have to remember to add them. This new method would make this a little easier - you can name all the properties you want to keep.

```
protected function resetExcept(...$properties)
{
    if (count($properties) && is_array($properties[0])) {
        $properties = $properties[0];
    }

    $keysToReset = array_diff(array_keys($this->getPublicPropertiesDefinedBySubClass()), $properties);
    $this->reset($keysToReset);
}
```

I don't know if that's of use for enough people, but I needed it. I have it in a base component anyway, so no hard feelings if this gets rejected :-).

**Comment by vmosoti** [comment]
Been needing this for quite many use cases. I hope it be considered.
